### PR TITLE
Overall clean up

### PR DESCRIPTION
**Pull request by muffato**
This is a bit of cleanup of the code and parameters that I want to get in before #97 and finally reaching v1.0.0.
The Changelog mentions a tentative intermediate v0.5.0 release, though I'm not sure it will happen at all. If I can get all the work done to support draft assemblies soon enough, I may ship both as v1.0.0.

- I've removed all the `--blast*` parameters because end users could not change them without breaking the pipeline.
- I've renamed `--busco` to prepare the introduction of the `--busco_lineages` parameter in #97 .
- I've removed `--taxa_file` because it didn't add anything more than `--taxon`.

- I've fixed the channel factories so that they're correctly queue/value channels, and paths are properly constructed as paths (files).
- Whitespace cleanup

And two pipeline changes:
- More resources for blastn
- Skip the blastn sub-workflow if there is nothing to blast

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/busco_diamond_blastp.nf [hunk 1]
```diff
@@ -108,11 +108,14 @@ workflow BUSCO_DIAMOND {
 
     //
     // Align BUSCO genes against the BLASTp database
-    //    
+    //
     BLOBTOOLKIT_EXTRACTBUSCOS.out.genes
     | filter { it[1].size() > 140 }
     | set { ch_busco_genes }
 
+    // Hardocded to match the format expected by blobtools
```
Comment (gq1) [inline subworkflows/local/busco_diamond_blastp.nf:116]:
```suggestion
    // Hardcoded to match the format expected by blobtools
```

**Comment by gq1** [comment]
Looks good for me.

How you picked up all the white spaces now? nf-core linting didn't complain it before?

I think it is ok to keep `--busco` for busco_db and add `busco_lineages`. It will be easier if we have lots of users. It is fine you have made all the changes.

**Review by gq1 — APPROVED** [review]

**Comment by muffato** [comment]
> I think it is ok to keep `--busco` for busco_db and add `busco_lineages`. It will be easier if we have lots of users. It is fine you have made all the changes.

Thanks for raising that. The introduction of `busco_lineages` is indeed the _only_ reason why I wanted to rename `--busco`. But here are some arguments for keeping `--busco`:

- backwards compatibility
- It would be the only database argument to be called `*_db`: `--blastp`, `--blastn`, `--blastx` and `taxdump` would remain without the `_db` suffix.
- probably only people who would set `--busco_lineages` would like the change. `--busco_lineages` is optional and the default behaviour (use all lineages that match the species) should work for many users already. These users would just use `--busco` and be ok with that

**Comment by muffato** [comment]
I decided to remove the commit that renamed the parameter

**Pull request merged**